### ISSUE TITLE
Patch: Speech with subtitles button in Audio Options gump - memory corruption problem

### ISSUE THREAD

**Issue by DominusExult** [exult/exult#136]
I'm trying to add a button to the Audio gump that enables the subtitles while speech is enabled. I'm doing something wrong with the rebuild button thing. The enabled button for the "subtitles" sometimes pulls in the text of other buttons (OK, HELP...) and after some opening and closing of the Audio gump it will crash.
I can't see what I'm doing wrong.

https://pastebin.com/KieJf4jv <- the patch

https://pastebin.com/816ypHYm <- the lldb bt

[Subtitles.diff.txt](https://github.com/exult/exult/files/7920495/Subtitles.diff.txt)

**Comment by Dragon-Baroque** [comment]
I cannot reproduce : Linux KDE, Exult branch master up to date.
The «subtitles» button correctly alternates between Disabled and Enabled, and properly disappears and reappears when the «Speech» button is toggled.

One surprise, though, is that the patch complained with small line changes :
```
$ patch -p1 <../Subtitles.diff 
patching file gumps/AudioOptions_gump.cc
Hunk #1 succeeded at 147 (offset 2 lines).
Hunk #2 succeeded at 208 (offset 2 lines).
Hunk #3 succeeded at 355 (offset 2 lines).
Hunk #4 succeeded at 471 (offset 4 lines).
Hunk #5 succeeded at 577 (offset 4 lines).
patching file gumps/AudioOptions_gump.h
```

**Comment by DominusExult** [comment]
I did some changes in between, that's why it doesn't apply cleanly.

But can you try:
- have speech enabled
- subtitles enabled

Open and close the audio gump a couple of times (maybe even closing it with "ok")
 
It doesn't always corrupt or crash but sometimes even on restarting Exult and opening the Audio gump

**Comment by DominusExult** [comment]
And I think you need to be in a game, not in the main menu of Exult.

**Comment by DominusExult** [comment]
Fixed with 0e7d9dd8 - I did something wrong with loading the state of the subtitles.
